packtools/sps/models/funding_group.py

```diff
@@ -24,19 +61,39 @@ class FundingGroup:
     def __init__(self, xmltree):
         self._xmltree = xmltree
 
-    @property
-    def financial_disclosure(self):
-        # TODO separar os valores entre 'funding-source' e 'award-id'
-        for node in self._xmltree.xpath(".//fn-group/fn[@fn-type='financial-disclosure']"):
-            yield " ".join([item.strip() for item in node.xpath(".//text()") if item.strip()])
+    def fn_financial_information(self, special_chars_funding=[], special_chars_award_id=[]):
+        items = []
+        for fn_type in ('financial-disclosure', 'supported-by'):
+            funding_sources = []
+            award_ids = []
+            for nodes in self._xmltree.xpath(f".//fn-group/fn[@fn-type='{fn_type}']"):
+                for node in nodes.xpath('p'):
+                    text = xml_utils.node_plain_text(node)
+                    if any([char.isdigit() for char in text]):
```

Comment:
@Rossi-Luciano sugiro que crie uma função para substituir `any([char.isdigit() for char in text])`, porque a listcomprehension será executada integralmente.
Na função faça:

```python
def has_digit(text):
    # haverá no máx 10 iterações (isso é relavante quando estamos processando pelo menos 1 milhão de XML )
    for n in range(0,10):
        if n in text:
            return True
    return False
```

Reply:
OK, @robertatakenaka